src/store/cache/collections/utils/addUsersFromCollections.ts

```diff
@@ -2,16 +2,31 @@ import { makeUid } from 'utils/uid'
 import { UserCollection } from 'models/Collection'
 import { Kind } from 'store/types'
 import { reformat as reformatUser } from 'store/cache/users/utils'
-import { put } from 'redux-saga/effects'
+import { put, select } from 'redux-saga/effects'
 import * as cacheActions from 'store/cache/actions'
+import { getAccountUser } from 'store/account/selectors'
+import { uniqBy } from 'lodash'
 
+/**
+ * Adds users from collection metadata to cache.
+ * Dedupes users and removes self.
+ * @param metadataArray
+ */
 export function* addUsersFromCollections(metadataArray: Array<UserCollection>) {
-  const users = metadataArray.map(m => ({
+  const accountUser: ReturnType<typeof getAccountUser> = yield select(
+    getAccountUser
+  )
+  const currentUserId = accountUser?.user_id
+  let users = metadataArray.map(m => ({
     id: m.user.user_id,
     uid: makeUid(Kind.USERS, m.user.user_id),
     metadata: reformatUser(m.user)
   }))
 
+  // Removes duplicates and self
```

Comment:
perhaps if currentUserId { } around the filter so for non-logged in users we can short circuit the logic

Reply:
yeah that's better